support/android/apk/buildSrc/src/main/kotlin/VersionCode.kt

```diff
@@ -0,0 +1,20 @@
+import java.text.SimpleDateFormat
+import java.util.Date
+
+// Generate unique version code based on the build date and time to support nightly
+// builds.
+//
+// The version scheme is currently: yyyyMMddXX
+// where
+// yyyy is the 4 digit year (e.g 2024)
+// MMdd is the 2 digit month followed by 2 digit day (e.g 0915 for September 15)
+// XX is currently hardcoded to 00, but can be used to distinguish multiple builds within
+// the same day.
+//
+// TODO: check if this interferes with caching of local builds and add option to use
+// a static version.
+val today = Date()
+
+val versionCodeString: String = SimpleDateFormat("yyyyMMdd00").format(today)
+
+val generatedVersionCode = versionCodeString.toInt()
```

Comment:
Does Gradle guarantee that the logic in `buildSrc` will be run only once for both servoapp and servoview. My concern is that `servoapp` and `servoview` artifacts produced by a single build might not have the same version if that is not garaunteed.

Reply:
From what I understand from experience and [documentation](https://docs.gradle.org/current/userguide/sharing_build_logic_between_subprojects.html), and [member](https://discuss.gradle.org/t/buildsrc-vs-build-logic/46708/2) of the gradle team, Yes it is only produced once.

Reply:
Thanks for the links! It does look like Gradle guarantees that the `buildSrc` is **built once** but I am not sure if that also implies if the built code **executes** once, globally across all sub-projects. In any case, we should be safe for now as the version code only has a granularity of a day, but this might be something we need to keep in mind if we ever extend it to include minutes, seconds or some other transient state.

Reply:
because the version code is assigned to a val, we can assume it compiles into a constant int and said class is just imported by other projects.

I'll resolve this for now.